tests/dataframe_protocol_test.py

```diff
@@ -369,3 +394,66 @@ def test_smoke_get_buffers_on_categorical_columns(df_factory):
     interchange_df = df.__dataframe__()
     interchange_col = interchange_df.get_column_by_name('x')
     interchange_col.get_buffers()
+
+
+@pytest.mark.xfail()
+def test_interchange_pandas_string_column():
+    import pandas as pd
+    data = ["foo", "bar"]
+    try:
+        from pandas.api.interchange import from_dataframe
+    except ImportError:
+        pytest.skip(f"pandas.api.interchange not found ({pd.__version__=})")
+    pd_df = pd.DataFrame({"x": pd.Series(data, dtype=pd.StringDtype())})
+    pd_interchange_df = pd_df.__dataframe__()
+    vaex_df = _from_dataframe_to_vaex(pd_interchange_df)
+    assert vaex_df["x"].tolist() == data
+
+
+def test_string_buffers(df_factory):
+    data = ["foo", "bar"]
+    x = np.array(data, dtype="U8")
+    df = df_factory(x=x)
+    if isinstance(df["x"].values, pa.lib.ChunkedArray):
+        pytest.xfail()
+    interchange_df = df.__dataframe__()
+    roundtrip_df = _from_dataframe_to_vaex(interchange_df)
+    assert roundtrip_df["x"].tolist() == data
+
+
+@pytest.mark.parametrize(
+    "labels", [[10, 11, 12, 13], ["foo", "bar", "baz", "qux"]]
+)
+def test_describe_categorical(df_factory, labels):
+    # See https://github.com/vaexio/vaex/issues/2113
+    data = [3, 1, 1, 2, 0]
+    x = np.array(data)
+    df = df_factory(x=x)
+    df = df.categorize('x', labels=labels)
+    interchange_df = df.__dataframe__()
+    interchange_col = interchange_df.get_column_by_name('x')
+    catinfo = interchange_col.describe_categorical
+    assert isinstance(catinfo, dict)
+    assert isinstance(catinfo["is_ordered"], bool)
+    assert isinstance(catinfo["is_dictionary"], bool)
+    assert catinfo["is_dictionary"]
+    assert isinstance(catinfo["categories"], _VaexColumn)
+    assert catinfo["categories"]._col.tolist() == labels
+
+
+@pytest.mark.parametrize(
+    "labels", [[10, 11, 12, 13], ["foo", "bar", "baz", "qux"]]
+)
+def test_interchange_categorical_column(df_factory, labels):
+    data = [3, 1, 1, 2, 0]
+    x = np.array(data)
+    df = df_factory(x=x)
+    if isinstance(df["x"].values, pa.lib.ChunkedArray):
+        pytest.xfail()
+    df = df.categorize('x', labels=labels)
+    interchange_df = df.__dataframe__()
+    interchange_col = interchange_df.get_column_by_name('x')
+    roundtrip_df = _from_dataframe_to_vaex(interchange_df)
+    data_as_labels = [labels[i] for i in data]
+    assert roundtrip_df["x"].values.tolist() in [data, data_as_labels]  # TODO
```

Comment:
I wasn't sure if the values should be the codes (`[3, 1, 1, 2, 0]`), or the the codes-as-their-labels (e.g. `["qux", "bar", "bar", "baz", "foo"]`).

Reply:
Good question, for vaex v4 it can be inconsistent (dict encoded probably returns the ['foo'], while a 'vaex-our-own-encoding' can return the indices instead. I've checked, and the current code path does seem to go via an arrow array, so this behaves as I want with data_as_labels (I'll push a commit)